### PR TITLE
Add name to DeveloperApplication 

### LINKED ISSUE #75: Add and utilize Developer Application names
## Summary

Developer applications should have a name that can be displayed to users to make it clear what they are for.

## Acceptance Criteria

1. Developer Applications have a `name` field in `schema.prisma` a default should exist for migrating existing developer applications in production
2. The developer application name is displayed along with the UUID on the `/ui/applications` page

### PR DESCRIPTION
**Pull request by Lopkop**
Hello there! This small PR closes #75 

1. Add name to DevApplication schema (the default is "Application")
2. Add name field in the endpoints
3. Refactor html file to display the name and id of application

### CODE REVIEW

File: abandonauth/routers/ui.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ async def list_developer_applications(request: Request) -> Response:
         "developer_apps.html",
         {
             "request": request,
-            "dev_apps": [x["id"] for x in dev_apps],
+            "dev_apps": [(x["name"], x["id"]) for x in dev_apps],
```
Comment (GDWR) [inline abandonauth/routers/ui.py:138]: I wonder if we could use a dictionary structure here, would make the HTML template usage clearer too.

```suggestion
            "dev_apps": [{"name": x["name"], "id": x["id"])} for x in dev_apps],
```

Or even just
```suggestion
            "dev_apps": [x for x in dev_apps],
```
although I prefer the first as we explicitly say what is available to the template.

**Review by GDWR — APPROVED** [review]
Looks good, just one small thing.

File: abandonauth/routers/ui.py (hunk 1 above)
Comment (Lopkop) [reply abandonauth/routers/ui.py]: Thanks for review!

**Review by fisher60 — CHANGES_REQUESTED** [review]
This looks really good. 

We need to update the template `create_developer_app.html` to use the new `name` field. Currently if you try to create a new developer app with the UI it will fail with an HTTP 422 error since the `name` is required now in the body.

**Comment by fisher60** [comment]
I just pushed the changes needed to use the UI for creating a developer app with a name. I also cleaned some things up.

I threw away some changes that appear to be from auto-formatting from another PR as they greatly complicated rebasing.

In hindsight, I should not have rebased and force-pushed since the history is kinda goofy and makes it look like I was more involved than I was. Rest assured; all credit to @Lopkop for implementing this feature.

**Pull request merged**